src/instructions/mod.rs

```diff
@@ -10,9 +10,34 @@ pub mod tables;
 pub mod tlb;
 
 /// Halts the CPU until the next interrupt arrives.
-#[inline(always)]
+#[inline]
 pub fn hlt() {
     unsafe {
         asm!("hlt" :::: "volatile");
     }
 }
+
+/// Emits a 'magic breakpoint' instruction for the [Bochs](http://bochs.sourceforge.net/) CPU
+/// emulator. Make sure to set `magic_break: enabled=1` in your `.bochsrc` file.
+#[inline]
+pub fn bochs_breakpoint() {
+    unsafe {
+        asm!("xchgw %bx, %bx" :::: "volatile");
+    }
+}
+
+/// Gets the current instruction pointer. Note that this is only approximate as it requires a few
+/// instructions to execute.
+#[inline]
```

Comment:
This function should probably be `inline(always)`. Without `always`, this is just a hint to the compiler that this might be useful for inlining. But this function does not make much sense without inlining, as it would always return the same value independent of the calling location.